lib/ansible/executor/process/worker.py

```diff
@@ -99,15 +119,16 @@ def start(self):
         make sure it is closed in the parent when start() completes.
         """
 
-        self._save_stdin()
         # FUTURE: this lock can be removed once a more generalized pre-fork thread pause is in place
         with display._lock:
-            try:
-                return super(WorkerProcess, self).start()
-            finally:
-                self._new_stdin.close()
-
-    def _hard_exit(self, e):
+            super(WorkerProcess, self).start()
+        # Since setsid is called later, if the worker is termed
+        # it won't term the new process group
```

Comment:
Is this for the case where an additional new process(es) is created from the fork (thus added into the process group)? If it is just the worker process in the group, the group should be terminated automatically on `SIGTERM`, no?

Reply:
`setsid` makes the process a leader, and calls `setpgrp` which makes the fork/child no longer part of the parent process group. So the fork, and all of it's children do not see the signal sent to the parent.